packages/studio/src/store/slices/pages/createPageSlice.ts

```diff
@@ -160,6 +160,82 @@ export const createPageSlice: SliceCreator<PageSlice> = (set, get) => {
         (component) => component.uuid === activeComponentUUID
       );
     },
+    addSelectedComponentUUID: (selectedUUID: string) => {
+      set((store) => {
+        store.selectedComponentUUIDs.add(selectedUUID);
+      });
+    },
+    addSelectedComponentRect: (
+      selectedUUID: string,
+      rect: DOMRectProperties
+    ) => {
+      set((store) => {
+        store.selectedComponentRectsMap.set(selectedUUID, rect);
+      });
+    },
+    clearSelectedComponents: () => {
+      set((store) => {
+        store.selectedComponentUUIDs = new Set();
+        store.selectedComponentRectsMap = new Map();
+      });
+    },
+    /**
+     * Adds all of the components and their children ordered between the selected component and
+     * the active component (inclusive) to selectedComponentUUIDs.
+     */
+    addShiftSelectedComponentUUIDs: (selectedComponent: ComponentState) => {
+      const selectedUUID = selectedComponent.uuid;
+      const {
+        activeComponentUUID,
+        getActivePageState,
+        addSelectedComponentUUID,
+      } = get();
+      const activePageState = getActivePageState();
+      const componentTree = activePageState?.componentTree;
+      if (selectedUUID === activeComponentUUID) return;
+      if (!componentTree) {
+        throw new Error(
+          `Error adding components: component tree is undefined.`
+        );
+      }
+      if (!activeComponentUUID) {
+        throw new Error(
+          `Error adding components: active component is undefined.`
+        );
+      }
+      const targetComponentUUIDs = [selectedUUID, activeComponentUUID] as [
```

Comment:
I think this syntax is a little funky?  If you want to keep it as a tuple so you can use `includes` and the spread operator later, I think you can set the type to be `targetComponentUUID: [string, string]`, but IDK if that's preferable (@oshi97/@nmanu1).  You can also just keep them separate, so you don't have do any extra type stuff at all!

Reply:
changed to `targetComponentUUID: [string, string]`! I like having them together bc I think it makes the code cleaner but up to to @oshi97/@nmanu1

Reply:
talked to nidhi and changed to `const targetComponentUUIDs = [selectedUUID, activeComponentUUID] as const;`